container/cli.py

```diff
@@ -94,63 +94,34 @@ def subcmd_help_parser(parser, subparser):
 
 def subcmd_push_parser(parser, subparser):
     subparser.add_argument('--username', action='store',
-                           help=(u'Username to log into registry. If not provided, '
-                                 u'it is expected that your ~/.docker/config.json '
-                                 u'contains your login information.'),
+                           help=u'If authentication with the registry is required, provide a valid username.',
                            dest='username', default=None)
     subparser.add_argument('--email', action='store',
-                           help=(u'Email to log into the registry. If not provided, '
-                                 u'it is expected that your ~/.docker/config.json '
-                                 u'contains your login information.'),
+                           help=(u'If authentication with the registry requires an email address, provide a '
+                                 u'valid email address'),
                            dest='email', default=None)
     subparser.add_argument('--password', action='store',
-                           help=(u'Password to log into registry. If not provided, '
-                                 u'you will be prompted for it.'),
+                           help=u'If authentication with the registry is required, provide a valid password.',
                            dest='password', default=None)
     subparser.add_argument('--url', action='store',
-                           help=(u'Base URL for your registry. If not provided, '
-                                 u'Docker Hub will be used.'),
+                           help=(u'When authenticating with a registry, provide the registry URL. Otherwise, the '
```

Comment:
Isn't this just an either/or with `--push-to`? What if we had `--push-to` accept a URL or the name?